NDTensors/src/UnallocatedArrays/src/unallocatedzeros.jl

```diff
@@ -0,0 +1,51 @@
+## TODO still working to make this implementation simplified
+struct UnallocatedZeros{ElT,N,Axes,Alloc<:AbstractArray{ElT,N}} <:
+       FillArrays.AbstractZeros{ElT,N,Axes}
+  z::FillArrays.Zeros{ElT,N,Axes}
+
+  function UnallocatedZeros{ElT,N,Axes,Alloc}(inds::Tuple) where {ElT,N,Axes,Alloc}
+    z = FillArrays.Zeros(inds)
+    ax = typeof(FillArrays.axes(z))
+    return new{ElT,N,ax,Alloc}(z)
+  end
+end
+
+data(Z::UnallocatedZeros) = Z.z
+copy(Z::UnallocatedZeros) = typeof(Z)(size(Z))
+Base.vec(Z::UnallocatedZeros) = typeof(Z)(length(Z))
```

Comment:
I don't think this definition makes sense since `typeof(Z)` won't necessarily have `ndims` of 1.

A different pattern for this in terms of generic functions could be:
```julia
vec(Z::UnallocatedZeros) = set_alloctype(parent(Z), set_ndims(alloctype(Z), 1))
```